Delta Course (#256)
mitre/saf-training: src/courses/delta/README.md

---
order: 01
next: 02.md

> title: Delta Process Developer Class

Reviewer: ```suggestion
title: 1. Delta Process Developer Class
```

author: Daniel Medina, George Dias
---

## 1.1 Class Overview

The purpose of this class is to demonstrate how the MITRE SAF Delta tool can be used to facilitate changes to InSpec profile baselines. This process is often utilized to create a new baseline profile from older revisions of the same baseline.

> ### 1.1.1 Class Objectives

Reviewer: ```suggestion
## 1.2 Class Objectives
```

I recommend only having a certain level of index if there will be more than one. For example try to only make a 1.1 if there will be a 1.2 or a 1.1.1 only if there will be a 1.1.2

By the end of this class, you should be able to:

> - Identify the requirements for conducting the delta process

Reviewer: ```suggestion

```

> - Understand the purpose and functionality of the SAF Delta process tools: `update_controls4delta` and `delta`

Reviewer: ```suggestion
- Understand the requirements, purpose, and functionality of the SAF Delta process tools: `update_controls4delta` and `delta`
```

> - Utilize these tools to update InSpec profiles

Reviewer: ```suggestion
- Update existing InSpec profiles to newer guidance documentation using `update_controls4delta` and `delta`
```

- Determine how and where to apply these tools in various scenarios (Use Cases)
- Use the delta fuzzy matching system to map controls between two profiles

> - Become familiar with the delta process logging capabilities and the artifacts it generates

Reviewer: ```suggestion
- Identify and review delta process history with generated logs and artifacts
```

> - Format InSpec profile controls appropriately before and/or after executing delta commands

Reviewer: I would add a What is Delta and Why are we talking about it idea, like the InSpec 101 from the developer class on page 1 of the beginner developer class or like the goal of the SAF on page 2 of the User class. We need to introduce the WHAT and WHY before we start talking about the HOW in page 2 here.
